### PR TITLE
feat(storage): support for encryption at rest flag

### PR DESCRIPTION
**Pull request by villevsv-upcloud**

### CODE REVIEW

File: upcloud_api/storage.py [hunk 1]
```diff
@@ -153,6 +168,9 @@ def to_dict(self):
                 dict_labels.append(label.to_dict())
             body['labels'] = dict_labels
 
+        if hasattr(self, 'enrypted') and isinstance(self.enrypted, bool):
+            body['server']['enrypted'] = "yes" if self.enrypted else "no"
```
Comment (scop) [inline upcloud_api/storage.py:172]: "enrypted" typos?

File: upcloud_api/storage.py [hunk 2]
```diff
@@ -55,12 +56,26 @@ def _reset(self, **kwargs) -> None:
         elif 'storage_size' in kwargs:
             self.size = kwargs['storage_size']
 
+        if 'encrypted' in kwargs and kwargs['encrypted'] == 'yes':
```
Comment (scop) [inline upcloud_api/storage.py:59]: Style/nit: could simplify to 
```suggestion
        if kwargs.get('encrypted') == 'yes':
```

Comment (villevsv-upcloud) [reply upcloud_api/storage.py]: :+1:

File: upcloud_api/storage.py (hunk 1 above)
Comment (villevsv-upcloud) [reply upcloud_api/storage.py]: Thanks! :sweat_smile: Not something that the tests would have caught.

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,7 +9,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
-----
+### Added
+
+- Storage encryption at rest support via `encrypted`
```
Comment (Darep) [inline CHANGELOG.md:14]: Maybe rephrase a bit so changelog can be used to understand where & what & how:

```suggestion
- Added `encrypted` boolean flag to Storage for encryption at rest support
```

File: docs/Storage.md [hunk 4]
```diff
@@ -51,7 +51,8 @@ storage1 = manager.create_storage(
     zone='fi-hel1',
     size=10,
     tier="maxiops",
-    title="my storage disk"
+    title="my storage disk",
+    encrypted=True
```
Comment (Darep) [inline docs/Storage.md:55]: Maybe shouldn't suggest enabling encryption by default in examples as it can slow down things and user might be confused why 🤔 

```suggestion
    encrypted=False
```

File: upcloud_api/cloud_manager/storage_mixin.py [hunk 5]
```diff
@@ -53,13 +54,20 @@ def create_storage(
         """
         if backup_rule is None:
             backup_rule = {}
+
+        if encrypted:
+            encrypted_str = 'yes'
+        else:
+            encrypted_str = 'no'
```
Comment (Darep) [inline upcloud_api/cloud_manager/storage_mixin.py:61]: code style nitpick: i like inline if 😄 

```suggestion
        encrypted_str = 'yes' if encrypted else 'no'
```
